### DISCUSSION TITLE
Hide atk-ua-wrap and atk-link inside atk-sidebar?

### DISCUSSION THREAD

**Discussion by bibicadotnet** [ArtalkJS/Artalk#645] [Q&A]
![2023-11-17_14-52-13](https://github.com/ArtalkJS/Artalk/assets/119997727/6eeb66fa-bd06-4602-9df6-ff8cf7c4260f)

Is there any way to hide or turn off the two values atk-ua-wrap and atk-link inside atk-sidebar?

```
.atk-ua-wrap {
display: none !important;
}
.atk-link {
display: none !important;
}
```
Outside the comment it works, but inside the atk-sidebar it doesn't work

**Comment by qwqcode** [comment]
Regarding the UA badge for comments, you can find the option under `Control Center` -> `Settings` -> `UI Settings` -> `UA Badge`, where you can control its display. It's unnecessary to hide it using CSS.

Currently, the interface of the sidebar is an independent iframe page (built with Vue). 😢 Unfortunately, there's no provision for inserting custom JS or CSS into the sidebar page at the moment, but this functionality will be added in the future. 😁

**Reply by bibicadotnet** [reply]
Ah ah, I misunderstood the UA Badge :D I thought it was the Badge Text and Badge Color (Color Hex) feature. I don't understand why it displays the user's browser and operating system information?

I want to hide the atk-link information inside the sidebar, because I don't want users to fill in junk links. When they accidentally click on their nick and avatar, it will go to unwanted pages.

Is there any way to solve this problem?

Can disabled `<a href=>` in atk-avatar and atk-nick?

**Reply by qwqcode** [reply]
Furthermore, if you wish to insert custom JS or CSS into the sidebar now, there is a more geeky approach. You can use web servers like Nginx or Caddy to reverse-proxy Artalk's backend server and separately modify a path `{API}/sidebar/index.html`, altering it to your customized HTML. You can download the original `index.html`, manually insert CSS and JS tags, although this process might be a bit cumbersome.

On the other hand, you can manually modify and compile the entire project. Refer to: https://github.com/ArtalkJS/Artalk/blob/master/CONTRIBUTING.md for guidance.

**Reply by qwqcode** [reply]
We do indeed need to add a configuration option to disable user input of website links, as well as provide a configuration option for injecting custom JS and CSS scripts into the sidebar. These new features will be added in the future.

**Reply by bibicadotnet** [reply]
Yes, I will try it and see how it goes

Thanks

**Comment by bibicadotnet** [comment]
> Can disabled `<a href=>` in atk-avatar and atk-nick?

The purpose of hiding .atk-links in my sidebar is mainly to avoid spam links. Currently, the solution via reverse-proxy or self-building is not possible because it is a bit complicated.

I think the simplest way is to turn off <a href=> in 2 locations, atk-avatar and atk-nick, which can also be a temporary solution.

After some Googling, I found a suitable solution

```
.atk-comment>.atk-avatar img {
 pointer-events: none !important;
 cursor: default !important;
}

.atk-item.atk-nick a {
 pointer-events: none !important;
 cursor: default !important;
}
```
The CSS code needs to address the active class and explain to the browser that the event upon the link click, should be **none**

Pretty good
